packages/components/src/components/select-wrapper/select-wrapper-dropdown/select-wrapper-dropdown-utils.ts

```diff
@@ -156,6 +154,12 @@ export const getHighlightedOptionMap = (arr: OptionMap[]): OptionMap => arr.find
 export const getValidOptions = (options: OptionMap[]): OptionMap[] =>
   options.filter((item) => !item.hidden && !item.initiallyHidden && !item.disabled);
 
+export const getLengthOfVisibleOptionsAndOptgroups = (options: OptionMap[]): number => {
+  const visibleItems = options.filter((item) => !item.hidden && !item.initiallyHidden).length;
+  const itemsWithOptgroups = options.filter((item) => item.title).length;
```

Comment:
Instead of filtering and iterating over all options twice, the filter conditions should be combined.  
Also with the current implementation an option is counted two times when it is not hidden and has a title?
Perhaps also rename function to something like `getAmount..`.

Reply:
"Instead of filtering and iterating over all options twice, the filter conditions should be combined."
--> yes, I tried it with `reduce` but didn't get the requested results. Any idea how to simplify it?

"Also with the current implementation an option is counted two times when it is not hidden and has a title?"
--> this is intended, because every "title" should render an additional `optgroup`-like element which must be also taken into account of the height calculation.

Reply:
Found a solution with `reduce`. Seems to be more solid.